### PR TITLE
Account Request workflow additions

### PR DESCRIPTION
**Pull request by tommckennon**
1) Prevent users from submitting multiple requests. (#66)
2) Delete non-approved account requests from accounts list #63
3) Added multi-select autocomplete component (tag-input.component), and implemented it for the field-coord-er-request-form.component;'s agol groups selection. #74
4) is_existing_account field added and set when creating new account requests (#80).
5) Add ability to disable a ResponseProject and notify sponsors/delegates (#51).
6) Update the RequestFieldCoordDialogComponent to pull the AGOLGroups that are set as is_auth_group (#55).
7) Send new project request emails to Coordinator Admin group members (#81)
8) Searching added.

Additional:
Additional unit testing added.
 
9) #101

### CODE REVIEW

**Comment by tbock** [comment]
@tommckennon I reworked a bunch of the tests to better show how to test these things.  I've significantly reworked the approval so please manually test as well as see if there are other automated tests we can implement.

File: accounts/tests.py [hunk 1]
```diff
@@ -127,23 +156,156 @@ def test_account_rejection(self, mock_post, mock_get):
...
+        sponsor = User.objects.get(pk=1)
+        sponsors = User.objects.filter(agol_info__delegates=sponsor)
+        sponsor_results = AGOLGroup.objects.filter(Q(response__users=sponsor) | Q(response__users__in=sponsors))
+        self.assertTrue(len(sponsor_results) > 0)
+
+
+class TestSponsor(TestCase):
+    fixtures = ['fixtures.json']
 
+    # Get Sponsors by param tests #
```
Comment (tbock) [inline accounts/tests.py:302]: @tommckennon what are you testing here exactly? This really looks like its just testing django/drf code.  If you want to just make sure a certain field is set in search_fields for that viewset... do that directly... do not testing django/drf

File: accounts/tests.py [hunk 2]
```diff
@@ -127,23 +156,156 @@ def test_account_rejection(self, mock_post, mock_get):
...
+        if 'is_auth_group' in os.environ:
+            is_auth_group = os.environ['is_auth_group']
+            if is_auth_group == 'true':
+                is_auth_group = True
+            elif is_auth_group == 'false':
+                is_auth_group = False
+            is_auth_group_results = list(AGOLGroup.objects.filter(is_auth_group=is_auth_group))
+            self.assertIsNotNone(is_auth_group_results)
+
+    def test_get_agol_groups_by_sponsor(self):
```
Comment (tbock) [inline accounts/tests.py:292]: @tommckennon what are you testing here?

File: accounts/tests.py [hunk 3]
```diff
@@ -127,23 +156,156 @@ def test_account_rejection(self, mock_post, mock_get):
...
+
+    # Get AGOLGroups by param tests #
+    @patch.dict('os.environ', {'all': 'true'})
+    def test_get_agol_groups_by_all_param(self):
+        if 'all' in os.environ:
+            if os.environ.get('all', 'false') == 'true':
+                all_results = list(AGOLGroup.objects.all())
+                self.assertIsNotNone(all_results)
+
+    @patch.dict('os.environ', {'is_auth_group': 'true'})
```
Comment (tbock) [inline accounts/tests.py:281]: @tommckennon why are you patching os.environ.  If you just want to create a dict... just create it in the test.  Also not sure what you are testing... again it looks like testing django code.

File: accounts/tests.py [hunk 4]
```diff
@@ -127,23 +156,156 @@ def test_account_rejection(self, mock_post, mock_get):
...
+            result = IsSponsor().has_object_permission(request, SponsorsViewSet, ac)
+            self.assertTrue(result)
+
+
+class TestAGOLGroups(TestCase):
+    fixtures = ['fixtures.json']
+
+    # Get AGOLGroups by param tests #
+    @patch.dict('os.environ', {'all': 'true'})
+    def test_get_agol_groups_by_all_param(self):
```
Comment (tbock) [inline accounts/tests.py:275]: @tommckennon this doesn't not appear to be testing anything real

File: ui/src/app/approval-list/approval-list.component.spec.ts [hunk 5]
```diff
@@ -1,25 +1,66 @@
...
+    injector = getTestBed();
+    httpMock = injector.get(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should return an Observable<string>', () => {
+    const expectedResponse = 'Accounts created. Existing account NOT updated.';
```
Comment (tbock) [inline ui/src/app/approval-list/approval-list.component.spec.ts:46]: @tommckennon The point of testing here is not to test the django rest api.  It is to test that the component reacts/does things based on certain responses from the api.  I see nothing in here that actually tests the component.

File: ui/src/app/approval-list/approval-list.component.spec.ts [hunk 6]
```diff
@@ -1,25 +1,66 @@
...
+    http.post(url, body).subscribe(resp => {
+      expect(resp).toEqual(expectedResponse);
+    });
+
+    const req = httpMock.expectOne(url);
+    expect(req.request.method).toBe('POST');
+    expect(req.request.url).toBe(url);
+    expect(req.request.body).toEqual(body);
+
+    req.flush(body);
```
Comment (tbock) [inline ui/src/app/approval-list/approval-list.component.spec.ts:62]: @tommckennon .flush is used to mock the api response.  A successful post to approve actually has a blank response.  Not sure what you are trying to do here.

File: ui/src/app/approval-list/approval-list.component.spec.ts [hunk 7]
```diff
@@ -1,25 +1,91 @@
...
+    matSnackBar = TestBed.get(MatSnackBar);
+    httpTestingController = TestBed.get(HttpTestingController);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should notify users of successful approval', () => {
```
Comment (tbock) [inline ui/src/app/approval-list/approval-list.component.spec.ts:73]: @tommckennon I reworked the tests for the approval list... plenty more could be used but I think these are good examples of testing certain things.  basically... does confirmApproval trigger the snackbar to return a certain message based on if it gets a good or error response from http client.

File: accounts/tests.py (hunk 1 above)
Comment (tommckennon) [reply accounts/tests.py]: Removed

File: accounts/tests.py (hunk 2 above)
Comment (tommckennon) [reply accounts/tests.py:278]: Removed

File: accounts/tests.py (hunk 3 above)
Comment (tommckennon) [reply accounts/tests.py]: Removed

File: accounts/tests.py (hunk 4 above)
Comment (tommckennon) [reply accounts/tests.py:264]: Just a thought on the override of the ViewSet's get_queryset. The class TestAGOLGroups is at the end of the tests module. If it's just taking up space, I'll remove it.

File: ui/src/app/approval-list/approval-list.component.spec.ts (hunk 5 above)
Comment (tommckennon) [reply ui/src/app/approval-list/approval-list.component.spec.ts]: I follow. Thank you.

File: ui/src/app/approval-list/approval-list.component.spec.ts (hunk 6 above)
Comment (tommckennon) [reply ui/src/app/approval-list/approval-list.component.spec.ts]: I follow.

**Comment by tbock** [comment]
@jlherring for 5 above should we only show users who were created with the account since that is now available as part of 4 above

**Comment by tbock** [comment]
Closing b/c next iteration is ready for staging and it includes all changes here and reworks some of them.

**Pull request closed without merging**